edb/edgeql/compiler/typegen.py

```diff
@@ -401,13 +401,7 @@ def type_to_typeref(
     include_children = (
         expr_type is s_types.ExprType.Update
         or expr_type is s_types.ExprType.Delete
-        or (
-            (
-                env.options.is_explain or
-                env.options.use_inheritance_ctes
-            )
-            and isinstance(t, s_objtypes.ObjectType)
-        )
+        or isinstance(t, s_objtypes.ObjectType)
```

Comment:
We can collapse out the `expr_type` conditions--those can only hold if the type is an object anyway.